content/src/main/content/jcr_root/apps/acs-commons/authoring/accordion/rte-accordion.js

```diff
@@ -0,0 +1,194 @@
+        },
+  
+        // Triggers on click of the RTE button
+        execute: function(id) {
+            // Relays this click event to the Command registered with this ID (in this case, set below in AccordionCmd)
+            this.editorKernel.relayCmd(id);
+        },
+  
+        // This marks the accordion icon as active or not based on the selected DOM element in selDef
+        updateState: function(selDef) {
+            // Returns true or false whether this plugin is active in the selected DOM element in selDef
+            var hasUC = this.editorKernel.queryState(RTEAccordion.FEATURE, selDef);
+
+            if ( this.accordionUI !== null ) {
+                // Set as selected (Adds a darker background color to the icon)
+                this.accordionUI.setSelected(hasUC);
+            }
+
+            if (hasUC === true) {
+                // Change the accordion icon to use the one with a plus to indicate that this adds another accordion item to the existing accordion
+                $("[data-action='" + RTEAccordion.GROUP + "#" + RTEAccordion.FEATURE + "']").removeClass("coral-Icon--feed").addClass("coral-Icon--feedAdd");
+            } else {
+                // Restore the original icon in case it was changed above
+                $("[data-action='" + RTEAccordion.GROUP + "#" + RTEAccordion.FEATURE + "']").removeClass("coral-Icon--feedAdd").addClass("coral-Icon--feed");
+            }
+        }
+    });
+
+    // Accordion Plugin Commands
+    RTEAccordion.AccordionCmd = new Class({
+        extend: CUI.rte.commands.Command,
+  
+        isCommand: function(cmdStr) {
+            return (cmdStr.toLowerCase() == RTEAccordion.FEATURE);
+        },
+  
+        getProcessingOptions: function() {
+            var cmd = CUI.rte.commands.Command;
+            return cmd.PO_SELECTION | cmd.PO_BOOKMARK | cmd.PO_NODELIST;
+        },
+  
+        // The base node for this plugin is a div with an accordion class, used in multiple places to check and add this element
+        _getTagObject: function() {
+            return {
+                "tag": "div",
+                "attributes": {
+                    "class" : RTEAccordion.FEATURE
+                }
+            };
+        },
+
+        // We don't want to add this plugin inside another element, this code will find the highest parent for the current context
+        _getHighestParent: function ( context, node ){
+            var common = CUI.rte.Common;
+            // Get the nearest parent of the passed node
+            var parent = common.getParentNode(context, node);
+
+            // If the returned parent has an accordion class, this is the higest parent for this accordion so return it
+            if ( parent.className == RTEAccordion.FEATURE ) {
+                return parent;
+            } 
+            // We don't want to go too far up the dom, if the next parent after this is null return this node
```

Comment:
Sorry, I think they're the result of my trying to understand everything that needs to happen in these classes. I haven't been able to find much in terms of documentation so this was my way of figuring it out. I can prune them down if they're hindering the readability.